src/script/cpp_api/s_base.h

```diff
@@ -102,6 +110,20 @@ class ScriptApiBase {
 	void setOriginDirect(const char *origin);
 	void setOriginFromTableRaw(int index, const char *fxn);
 
+	void clientOpenLibs(lua_State *L);
+	const std::unordered_map<std::string, lua_CFunction> m_libs = {
+		{ "", luaopen_base },
+		{ LUA_LOADLIBNAME, luaopen_package },
+		{ LUA_TABLIBNAME,  luaopen_table   },
+		{ LUA_OSLIBNAME,   luaopen_os      },
+		{ LUA_STRLIBNAME,  luaopen_string  },
+		{ LUA_MATHLIBNAME, luaopen_math    },
+		{ LUA_DBLIBNAME,   luaopen_debug   },
+#if USE_LUAJIT
+		{ LUA_JITLIBNAME,  luaopen_jit     },
+#endif
+	};
```

Comment:
This should be a static local variable in `clientOpenLibs`.  Also, you could use a simpler data structure like a `std::vector<std::pair<std::string, lua_CFunction>>` for this.